lib/languagetool/actions/check.rb

```diff
@@ -15,7 +15,14 @@ def run
       protected
 
       def query
-        normalize_query(PARAMETERS.map { |k| [k, options[k]] })
+        normalize_query(parameters_with_common_query_params)
+      end
+
+      def parameters_with_common_query_params
+        PARAMETERS.map { |k| [k, options[k]] } + [
+          [:username, api.username],
+          [:api_key, api.api_key]
+        ]
```

Comment:
These changes should belong to the base action `LanguageTool::Actions::Base`. You also shouldn't hardcode which parameters are sent, but let the coder specify them via `common_query_params`.

I think `normalize_query` can hold this logic, and you could simply do something like this:

````rb
def normalize_query(query)
  query = api.common_query_params.merge(query)

  # ...
end
````